go-controller/pkg/cni/cni.go

```diff
@@ -80,11 +82,11 @@ func (pr *PodRequest) String() string {
 // and the best we can do is use the given UID for the duration of the request.
 // But if the existing UID is valid and does not match the given UID then the
 // sandbox request is for a different pod instance and should be terminated.
-func (pr *PodRequest) checkOrUpdatePodUID(podUID string) error {
-	if pr.PodUID == "" {
-		// Runtime didn't pass UID, use the one we got from the pod object
-		pr.PodUID = podUID
-	} else if podUID != pr.PodUID {
+func (pr *PodRequest) checkOrUpdatePodUID(pod *kapi.Pod) error {
+	if pr.PodUID == "" || IsStaticPod(pod) {
+		// Runtime didn't pass UID, or the pod is a static pod, use the one we got from the pod object
```

Comment:
nit: Would it be god to leave a comment here for posterity?
```
// Static pod UID is a hash of the pod itself that does not match
// the UID of the mirror kubelet creates on the api /server. 
// We will use the UID of the mirror.
// The hash is annotated in the mirror pod (kubernetes.io/config.hash)
// and we could match against it, but let's avoid that for now as it is not 
// a published standard.
```

Reply:
yes i think this makes sense